coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,375 @@
+        """
+        example_language = 'python'
+        importance_reason = """
+        Having a specific number of blank lines after every classes'
+        definitions declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        """
+    blank_lines_after_class = Taste[int](
+        'Represents the number of blank lines to use after a class'
+        'definition.', (1, 2), default=2)
+
+
+@BlankLine.subaspect
+class NewlineAtEOF:
+    """Illustrate the end of file character.
+
+    A newline character (usually '\\\\n', aka CR or CRLF) found at the end
+    of file.
+    """
+    class docs:
+        example = """
+        def do_nothing():
+            pass\\n
+        """
+        example_language = 'python'
+        importance_reason = """
+        A text file consists of a series of lines, each of which ends with a
+        newline character (\\\\n). A file that is not empty and does not end
+        with a newline is therefore not a text file.
+
+        It's not just bad style, it can lead to unexpected behavior, utilities
+        that are supposed to operate on text files may not cope well with files
+        that don't end with a newline.
+        """
+        fix_suggestions = """
+        `NewlineAtEOF` issues can be fixed by simply adding a newline at the
+        end of the file.
+        """
+    newline_at_EOF = Taste[bool](
+        'If ``True``, enforce a newline at End Of File.',
+        (True, False), default=True)
+
+
+@Spacing.subaspect
+class SpacesAroundOperator:
+    """Spacing around operators.
+    """
+    class docs:
+        example = """
+        def f(a):
+            return 37+-+a[42-x :  y**3]
+        """
+        example_language = 'python'
+        importance_reason = """
+        Having a specific and reasonable number of whitespace (blank) around
+        operators improves on the readability of the code.
+        """
+        fix_suggestions = """
+        `SpacesAroundOperator` issues can be fixed by simply specifying (and
+        using) a number of whitespace to be used after each operator.
+        """
+    spaces_around_operators = Taste[int](
+        'Represents the number of space to be used around operators.',
+        (0, 1), default=1)
+    spaces_before_colon = Taste[int](
+        'Represents the number of blank spaces before colons.',
+        (0, 1), default=0)
+    spaces_after_colon = Taste[int](
+        'Represents the number of blank spaces after colons.',
+        (0, 1), default=1)
+
+
+@Spacing.subaspect
+class UseSpaces:
+    """Determines if spaces should be used over tabs.
+    """
+    class docs:
+        example = """
+        def func(a, b):
+            result = a**b
+        \\treturn result
+        """
+        example_language = 'python'
+        importance_reason = """
+        A tab could be a different number of columns depending on the
+        environment(text editor), but a space is always one column. For
```

Comment:
simplify `environment(text editor),`  - brackets shouldnt be needed to understand this.